lightning/src/chain/channelmonitor.rs

```diff
@@ -1437,71 +1601,12 @@ impl<Signer: Sign> ChannelMonitor<Signer> {
-							// HTLC transactions are resolved, unless we're talking about a holder
-							// commitment tx, whose resolution is delayed until the CSV timeout is
-							// reached, even though HTLCs may be resolved after only
-							// ANTI_REORG_DELAY confirmations.
-							debug_assert!($holder_commitment || us.funding_spend_confirmed.is_some());
-						} else if htlc.offered == $holder_commitment {
-							// If the payment was outbound, check if there's an HTLCUpdate
-							// indicating we have spent this HTLC with a timeout, claiming it back
-							// and awaiting confirmations on it.
-							let htlc_update_pending = us.onchain_events_awaiting_threshold_conf.iter().find_map(|event| {
-								if let OnchainEvent::HTLCUpdate { commitment_tx_output_idx: Some(commitment_tx_output_idx), .. } = event.event {
-									if commitment_tx_output_idx == htlc_commitment_tx_output_idx {
-										Some(event.confirmation_threshold()) } else { None }
-								} else { None }
-							});
-							if let Some(conf_thresh) = htlc_update_pending {
-								res.push(Balance::ClaimableAwaitingConfirmations {
-									claimable_amount_satoshis: htlc.amount_msat / 1000,
-									confirmation_height: conf_thresh,
-								});
-							} else {
-								res.push(Balance::MaybeClaimableHTLCAwaitingTimeout {
-									claimable_amount_satoshis: htlc.amount_msat / 1000,
-									claimable_height: htlc.cltv_expiry,
-								});
-							}
-						} else if us.payment_preimages.get(&htlc.payment_hash).is_some() {
-							// Otherwise (the payment was inbound), only expose it as claimable if
-							// we know the preimage.
-							// Note that if there is a pending claim, but it did not use the
-							// preimage, we lost funds to our counterparty! We will then continue
-							// to show it as ContentiousClaimable until ANTI_REORG_DELAY.
-							let htlc_spend_pending = us.onchain_events_awaiting_threshold_conf.iter().find_map(|event| {
-								if let OnchainEvent::HTLCSpendConfirmation { commitment_tx_output_idx, preimage, .. } = event.event {
-									if commitment_tx_output_idx == htlc_commitment_tx_output_idx {
-										Some((event.confirmation_threshold(), preimage.is_some()))
-									} else { None }
-								} else { None }
-							});
-							if let Some((conf_thresh, true)) = htlc_spend_pending {
-								res.push(Balance::ClaimableAwaitingConfirmations {
-									claimable_amount_satoshis: htlc.amount_msat / 1000,
-									confirmation_height: conf_thresh,
-								});
-							} else {
-								res.push(Balance::ContentiousClaimable {
-									claimable_amount_satoshis: htlc.amount_msat / 1000,
-									timeout_height: htlc.cltv_expiry,
-								});
-							}
+					if htlc.transaction_output_index.is_some() {
+
+						if let Some(bal) = us.get_htlc_balance(htlc, $holder_commitment, $counterparty_revoked_commitment, confirmed_txid) {
+							res.push(bal);
 						}
 					}
```

Comment:
I'd be in favor of adding an extra parameter to `get_htlc_balance` for the `htlc.transaction_output_index`, then you don't need to redundantly check that it's `Some` and it's clearer that `Balance` may not be set in `get_htlc_balance` even if it's non-dust

Reply:
Ah ok, good as is then